File: problem0010/main.hs
main :: IO ()
main = print . sum . primesBelow $ 2000000

primesBelow :: Int -> [Int]
primesBelow n = filter isPrime [2..n]

isPrime :: Int -> Bool
isPrime n = all (\d -> n `mod` d /= 0) dividers
  where
    dividers = [2..(floor . sqrt . fromIntegral $ n)]

File: problem0010/main.rs
fn main() {
    println!("{}", sum_of_primes_below(2_000_000));
}

fn sum_of_primes_below(n: u64) -> u64 {
    (2..=n).filter(|v| is_prime(*v)).fold(0, |sum, v| sum + v)
}

fn is_prime(n: u64) -> bool {
    for d in 2..=(n as f64).sqrt().floor() as u64  {
        if n % d == 0 {
            return false;
        }
    }
    true
}
